docs/sources/query/logcli/getting-started.md

```diff
@@ -926,28 +914,137 @@ Args:
+1. Be aware there are no **labels** when using `--stdin`. So the stream selector in the query is optional, for example, just `|="timeout"|logfmt|level="error"` is same as `{foo="bar"}|="timeout|logfmt|level="error"`
+
 {{< admonition type="note" >}}
-Currently it doesn't support any type of metric queries.
+Currently `stdin` doesn't support any type of metric queries.
 {{< /admonition >}}
 
-You may have to use `stdin` flag for several reasons
-1. Quick way to check and validate a LogQL expressions.
-2. Learn basics of LogQL with just Log files and `LogCLI`tool ( without needing set up Loki servers, Grafana etc.)
-3. Easy discussion on public forums. Like Q&A, Share the LogQL expressions.
-
-**NOTES on Usage**
-1. `--limits` flag doesn't have any meaning when using `--stdin` (use pager like `less` for that)
-1. Be aware there are no **labels** when using `--stdin`
-   - So stream selector in the query is optional e.g just `|="timeout"|logfmt|level="error"` is same as `{foo="bar"}|="timeout|logfmt|level="error"`
+#### `stdin` examples
 
-**Examples**
 1. Line filter - `cat mylog.log | logcli --stdin query '|="too many open connections"'`
-2. Label matcher - `echo 'msg="timeout happened" level="warning"' | logcli --stdin query '|logfmt|level="warning"'`
-3. Different parsers (logfmt, json, pattern, regexp) - `cat mylog.log | logcli --stdin query '|pattern <ip> - - <_> "<method> <uri> <_>" <status> <size> <_> "<agent>" <_>'`
-4. Line formatters - `cat mylog.log | logcli --stdin query '|logfmt|line_format "{{.query}} {{.duration}}"'`
+1. Label matcher - `echo 'msg="timeout happened" level="warning"' | logcli --stdin query '|logfmt|level="warning"'`
+1. Different parsers (logfmt, json, pattern, regexp) - `cat mylog.log | logcli --stdin query '|pattern <ip> - - <_> "<method> <uri> <_>" <status> <size> <_> "<agent>" <_>'`
+1. Line formatters - `cat mylog.log | logcli --stdin query '|logfmt|line_format "{{.query}} {{.duration}}"'`
+
+## Batching
+
+logcli sends queries to Loki in such a way that query results arrive in batches.
+
+The `--limit` option for a `logcli query` command limits the total number of log lines that will be returned for a single query.
+When not set, `--limit` defaults to 30.
+The limit protects the user from overwhelming Loki in cases where the specified query would have returned a large number of log lines.
+The limit also protects the user from unexpectedly large responses.
+
+Larger result sets can be batched for easier consumption. Use the `--batch` option to control the number of log line results that are returned in each batch.
+When not set, `--batch` defaults to 1000.
+
+Setting a `--limit` value larger than the `--batch` value will cause the
+requests from logcli to Loki to be batched.
+
+When you run a query in Loki, it will return up to a certain number of log lines. By default, this limit is 5000 lines. You can configure this server limit with the `limits_config.max_entries_limit_per_query` in Loki's configuration.
```

Comment:
```suggestion
When you run a query in Loki, it will return up to a certain number of log lines. By default, this limit is 5000 lines. You can configure this server limit with the `limits_config.max_entries_limit_per_query` in Loki's [configuration](https://grafana.com/docs/loki/<LOKI_VERSION>/configure/#limits_config).
```